src/test/server/store-test.js

```diff
@@ -0,0 +1,201 @@
+const assert = require('assert');
+
+const usersData  = require('../../data/users.json');
+const UserAction = require('../../../_compiled_/actions/user-action').default;
+const UserInfo   = require('../../../_compiled_/stores/user-info').default;
+const UserStore  = require('../../../_compiled_/stores/user-store').default;
+const EVENT_TYPE = require('../../../_compiled_/stores/event-type').default;
+
+const TEST_USER = new UserInfo(
+    {
+        id:      11,
+        name:    'Ervin New',
+        phone:   '89500418181',
+        email:   'vika.chernookaya@mail.ru',
+        website: 'github.com'
+    }
+);
+
+const TEST_USER_EXISTING = new UserInfo(
+    {
+        id:      11,
+        name:    'Ervin Howell',
+        phone:   '0418181',
+        email:   'ervin@mail.ru',
+        website: 'github.com'
+    }
+);
+
+const SEARCH_USER_INFO = new UserInfo(
+    {
+        name: 'Ervin'
+    }
+);
+
+const SEARCH_USER_INFO_2 = new UserInfo(
+    {
+        name:  'Ervin  ',
+        phone: '  '
+    }
+);
+
+const UPDATE_USER = new UserInfo(
+    {
+        id:      2,
+        name:    'Vika',
+        phone:   '555',
+        email:   'ervin@mail.ru',
+        website: 'github.com'
+    }
+);
+
+const SEARCH_FIELDS = {
+    name:    true,
+    phone:   false,
+    email:   false,
+    website: false
+};
+
+let users;
+let foundUsers;
+let searchFields;
+let userId;
+
+UserStore.on(EVENT_TYPE.usersFound, _getFoundUsers);
+UserStore.on(EVENT_TYPE.addingFailed, _getUserID);
+
+function _getFoundUsers (users, fields) {
```

Comment:
these method are quite hard to undersand and they look redundant
`_getFoundUsers`
`_getUserID`
We can substribe for `addingFailed` in the `'Should emit addingFailed event with duplicateUserId if user exists ` test.
We can substribe for `usersFound` globally but 
1) please don't forget to unsubsribe
2) your tests look like they are synchronous but getting events (`usersFound` or `addingFailed`) is async
we should use Promises or `done` methods